lib/uefi/boot_service_provider.cpp

```diff
@@ -0,0 +1,380 @@
+
+#include "boot_service_provider.h"
+#include "boot_service.h"
+#include "kernel/vm.h"
+#include "lib/dlmalloc.h"
+#include "types.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+EfiStatus unload(EfiHandle handle) { return SUCCESS; }
+
+bool guid_eq(const EfiGuid *a, const EfiGuid *b) {
```

Comment:
here and the rest of this code, if these are local functions they should be marked static or in an anonymous namespace to help the linker inline and GC them.

Reply:
Done